Corrected Sinh model's definition, init fit parameter & fold definition.

AndrewYongZhenNing opened the pull request:
Summary of 7/7/2020 PR

1. Previous definition of sinh model means fit parameter Z_0 <0 for a 'normal' sinh behaviour. New defn. of sinh model has appropriate alternate minus sign. 
2. Previous definition of fold function is only applicable for cosh(or cst) model. New defn of fold functio requires a second parameter: modelPar. This is evaluated in a 'switch' block that changes sign of the folding definition. 
3. Based on 1., corrected init(1) of sinh to have the correct alternate sign.

Commit:
1) fold now requires second argument: modelPar to properly fold correlators with cosh/sinh form. 2) makeSinhModel has the appropriate alternate sign s.t the prefactor remains positive. 3) parameterGuess uses correct analytic form for sinh-type correlator's init(1), corresponding to the prefactor.

Commit:
2pt-fit.cpp now has correct number of arguments for fold function if --fold is called.

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

aportelli reviewed (changes requested):
Thanks. As we discussed only keep changes related to folding.

Commit:
1) restored original definition of sinh in makeSinhModel() and its init in parameterGuess(). 2) fold definition includes sign factor due to cosh/sinh model.

Commit:
latan-sample-fake now takes option -r/--seed where seed can be specified if fake bootstrap samples with 100% correlation are needed.

AndrewYongZhenNing commented:
13/11/2020 Summary

1) Pending 7/7/2020 PR: Included a sign factor when folding cosh/sinh model in CorrelatorFitter.cpp and 2pt-fit.cpp
2) latan-sample-fake now takes optional argument -r/--seed if fake bootstrap samples with 100% correlation are required.

Commit:
New methods in XYSampleData class to skip fits that are non-converging/large chi2PerDof.
This is tracked by boolean goodFit_.

AndrewYongZhenNing committed:
Merge remote-tracking branch 'fork/develop' into develop

Commit:
Added feature to reject fit if 25% of sample fits are bad (with chi2perdof > 2 or is a NaN)

AndrewYongZhenNing committed:
Merge remote-tracking branch 'fork/develop' into develop

Commit:
Removed criteria of rejecting fits if sample fit chi2PerDof is bad.

AndrewYongZhenNing committed:
Merge remote-tracking branch 'fork/develop' into develop

Commit:
Previous commit failed to compile: needed to remove leftover lines.

AndrewYongZhenNing committed:
Merge remote-tracking branch 'fork/develop' into develop

Commit:
Added private variables to set bound on chi2perDof. This is used tas criteria for fit scans.
TODO: modify to exception handling.

AndrewYongZhenNing committed:
Merge remote-tracking branch 'fork/develop' into develop

AndrewYongZhenNing committed:
XYSampleData throws Runtime exception error if central fit's chi2PerDof exceeds user-set/default bounds for chi2PerDof. This can be used to skip fits with bad fitranges.

AndrewYongZhenNing committed:
Changed fit criteria from chi2PerDof to pValue.

AndrewYongZhenNing committed:
New function fitSample to make XYSampleData::fit more modular:
Eg application: In fit scans, one may only want central fit for the uncorr fit to speed up process.

AndrewYongZhenNing committed:
i) removed pvalue fit criteria
ii) overloaded fit() to output FitResult object of a particular sample

aportelli committed:
Update Readme.md

AndrewYongZhenNing committed:
strTo<> now converts string of numbers to vector<Index>

AndrewYongZhenNing committed:
Merge branch 'alt-fit-scan' into Tesseract

AndrewYongZhenNing committed:
corrected own bug in strTo<T>

AndrewYongZhenNing committed:
removed redundant pValue private member

aportelli committed:
first skeleton for DWT

aportelli committed:
DWT working and tested

AndrewYongZhenNing committed:
Merge from AP's develop.

AndrewYongZhenNing committed:
Modified fitSample routines to use SampleFitResult objects.

aportelli committed:
utility to compute sample DWT

aportelli committed:
data plots compatible with multi-column arrays

aportelli committed:
correlation matrix plot compute dynamic range

aportelli committed:
tool to merge samples

AndrewYongZhenNing committed:
Restructured sample-read to print stat err next to central value.

AndrewYongZhenNing committed:
Merge branch 'develop' of https://github.com/aportelli/LatAnalyze into develop

Commit:
Added histogram feature to return xMax and xMin.

Commit:
Merge branch 'develop' of https://github.com/AndrewYongZhenNing/LatAnalyze into develop

aportelli committed:
Merge branch 'andrew-pr' into develop
